platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/PostHashCollectorWiring.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring.components;
+
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.common.wiring.wires.output.OutputWire;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Wiring object that allows for the staging of events that have been hashed, but haven't been passed further down the
+ * pipeline yet.
+ * <p>
+ * This wiring object is a workaround for the problems that concurrent schedulers currently have with pushing to
+ * components that apply backpressure.
+ *
+ * @param eventInput  the input wire for events that have been hashed
+ * @param eventOutput the output wire for events to be passed further along the pipeline
+ */
+public record PostHashCollectorWiring(
```

Comment:
I don't quite understand the purpose of this. Some more explanation would be useful for me

Reply:
@lpetrovic05 I pushed an expanded description of the problem